src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs

```diff
@@ -72,6 +72,26 @@ public static void WriteMetricsCollection(this PrometheusExporter exporter, Stre
                             break;
                         }
 
+                    case MetricType.DoubleSum:
+                        {
+                            builder = builder.WithType(PrometheusCounterType);
+                            foreach (var metricPoint in metric.GetMetricPoints())
+                            {
+                                var metricValueBuilder = builder.AddValue();
+                                metricValueBuilder = metricValueBuilder.WithValue(metricPoint.DoubleValue);
+                                if (metricPoint.Keys != null)
```

Comment:
Is this check only necessary for `Counter`s and not `Gauge`s?

Reply:
Good catch! its needed for all. (I was hoping to move these to all common helpers, so don't have to repeat this check. But thats hasn't been done yet).
Feel free to submit a PR to fix the current issue. Else I'll send one. let me know if you are working on it.

Reply:
#2346